src/main/java/com/patina/codebloom/api/admin/AdminController.java

```diff
@@ -157,4 +158,27 @@ public ResponseEntity<ApiResponder<Announcement>> createNewAnnouncement(
 
         return ResponseEntity.ok(ApiResponder.success("New announcement successfully created!", announcement));
     }
+
+    @Operation(summary = "Create a delete announcement if exist (only for admins).", responses = {
+            @ApiResponse(responseCode = "401", description = "Not authenticated", content = @Content(schema = @Schema(implementation = UnsafeGenericFailureResponse.class))),
+            @ApiResponse(responseCode = "200", description = "Announcement successfully Deleted"),
+            @ApiResponse(responseCode = "500", description = "Something went wrong", content = @Content(schema = @Schema(implementation = UnsafeGenericFailureResponse.class)))
+    })
+    @PostMapping("/announcement/delete")
+    public ResponseEntity<ApiResponder<Announcement>> deleteAnnouncement(@Valid @RequestBody final DeleteAnnouncementBody deleteAnnouncementBody, final HttpServletRequest request) {
+        protector.validateAdminSession(request);
+        Announcement announcement = announcementRepository.getAnnouncementById(deleteAnnouncementBody.getId());
+        if (announcement == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, " announcement does not exist");
```

Comment:
nit: extra space + add caps
```suggestion
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Announcement does not exist");
```